src/main/java/org/reactome/release/qa/common/QACheckerHelper.java

```diff
@@ -273,192 +278,43 @@ public static String getAttributeTableName(String clsName,
-	 * @param multiPEInstance GKInstance -- Complex, Polymer or EntitySet instance that will be searched for all PhysicalEntity instances.
-	 * @return Set<GKInstance> -- All distinct PhysicalEntities that are found in Complex/Polymer/EntitySet.
-	 * @throws Exception -- Thrown by MySQLAdaptor.
-	 */
-	public static Set<GKInstance> findAllConstituentPEs(GKInstance multiPEInstance) throws Exception {
-		Set<GKInstance> physicalEntities = new HashSet<>();
-		if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.Complex) || multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.Polymer)) {
-			// Recursively search Complex/Polymer for all PhysicalEntities.
-			for (GKInstance constituentPE : getComplexOrPolymerConstituentPEs(multiPEInstance)) {
-				physicalEntities.addAll(findAllPhysicalEntities(constituentPE));
-			}
-		} else if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.EntitySet)) {
-			// Recursively search members and candidates (if EntitySet is a CandidateSet) for all PhysicalEntities.
-			physicalEntities.addAll(findEntitySetPhysicalEntities(multiPEInstance));
-		}
-		return physicalEntities;
-	}
-
-	/**
-	 * Helper method that returns either a Complexes' components or Polymers' repeatedUnits.
-	 * @param multiPEInstance GKInstance -- Either a Complex or Polymer instance.
-	 * @return Set<GKInstance> -- All components or repeatedUnits from the incoming instance.
+	 * This method utilizes the InstanceUtilities method for finding all reaction participants. It also checks for the
+	 * 'activeUnit' participant in catalysts/regulations that can be flagged by certain QA checks. Generally speaking,
+	 * the 'activeUnit' instance should be found in the 'physicalEntity' of catalysts/regulations. Sometimes this is not
+	 * the case, when the wrong activeUnit is mistakenly checked in. This will be flagged by QA tests that call this method.
+	 * @param reaction GKInstance -- ReactionlikeEvent that will be checked for all participants.
+	 * @return Set<GKInstance> -- All distinct participants in the incoming ReactionlikeEvent.
 	 * @throws Exception -- Thrown by MySQLAdaptor.
 	 */
-	private static Collection<GKInstance> getComplexOrPolymerConstituentPEs(GKInstance multiPEInstance) throws Exception {
-		Set<GKInstance> constituentPEs = new HashSet<>();
-		if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.Complex)) {
-			constituentPEs.addAll(multiPEInstance.getAttributeValuesList(ReactomeJavaConstants.hasComponent));
-		} else if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.Polymer)) {
-			constituentPEs.addAll(multiPEInstance.getAttributeValuesList(ReactomeJavaConstants.repeatedUnit));
-		}
-		return constituentPEs;
-	}
+	public static Set<GKInstance> getAllReactionParticipantsIncludingCatalystAndRegulations(GKInstance reaction) throws Exception {
+		Set<GKInstance> reactionPEs = new HashSet<>();
+		reactionPEs.addAll(InstanceUtilities.getReactionParticipants(reaction));
```

Comment:
Maybe this method could get renamed to something like `getReactionInputsAndOutputs`.  The name could be confusing or misleading if this doesn't include catalysts and regulators of the reaction.

Reply:
That doesn't quite capture what this method is trying to do, maybe I'll rename it to be 'IncludingActiveUnit' instead

Reply:
Changed it to say 'IncludingActiveUnits'